pulp_ansible/app/galaxy/v3/views.py

```diff
@@ -153,8 +153,26 @@ def get_queryset(self):
         deprecated_query = AnsibleCollectionDeprecated.objects.filter(
             collection=OuterRef("pk"), repository_version=repo_version
         )
-        collections = Collection.objects.filter(versions__in=repo_version.content).distinct()
-        collections = collections.annotate(deprecated=Exists(deprecated_query))
+        added_repo_version_query = RepositoryVersion.objects.filter(
+            repository=repo_version.repository,
+            complete=True,
+            added_memberships__content__in=OuterRef("versions__pk"),
+        ).values("pulp_last_updated")
+        removed_repo_version_query = RepositoryVersion.objects.filter(
+            repository=repo_version.repository,
+            complete=True,
+            removed_memberships__content__in=OuterRef("versions__pk"),
+        ).values("pulp_last_updated")
+
+        collections = (
```

Comment:
This line reads a bit confusing because I keep thinking it is a tuple definition but it's not. It's valid Python though. Consider maybe:

```
collections_qs = Collection.objects.filter(versions__in=repo_version.content).distinct()
collections_qs =  collection_qs.annotate(
    repo_version_added_at=Max(Subquery(added_repo_version_query)),
    repo_version_removed_at=Max(Subquery(removed_repo_version_query)),
    deprecated=Exists(deprecated_query),
)
```

```
collections_qs = Collection.objects.filter(versions__in=repo_version.content).distinct()
collections_qs_annotated =  collection_qs.annotate(
    repo_version_added_at=Max(Subquery(added_repo_version_query)),
    repo_version_removed_at=Max(Subquery(removed_repo_version_query)),
    deprecated=Exists(deprecated_query),
)
```

Reply:
I also didn't like it, it was black, I'll improve it